git_hooks/pre-commit

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env sh
```

Comment:
I skipped these steps a couple of times, I think it's useful to provide this an an optional utility.

Reply:
I don't know if this is something for us to provide, I know that I already have hooks running (plus you need to run clippy before check, it's a known issue)

Reply:
@Kixiron regarding the order, if I'll first run `clippy` and then run `fmt` I won't get the warning? So we can remove the `dead_code` that I added?

Also, it's an **optional** utility to make the contributor local  environment match the CI/CD, I actually find these sort of helpers missing from various projects and as you mentioned yourself, we all end up recreating them locally which is literally the opposite intention an open source project like this one.

Reply:
If you run `cargo check` before `cargo clippy` it doesn't output anything, even when there's lints (usually) and emoving the dead code attribute comes from fixing the problem it's talking about

Reply:
+1 to having pre-commit hooks, otherwise you end up with people having to force-push three times because they forgot to run `cargo fmt`. That said I think this should be either a hard or soft link so that it doesn't get outdated when the commit script is updated: `ln -s ../../git_hooks/pre-commit .git/hooks` (not sure the windows equivalent - maybe `mklink` @Kixiron ?). Also, does it really need to be in a subdirectory if it's only one file?

Reply:
It doesn't really matter what command you run, git runs hooks in a bash environment

Reply:
Right, I'm asking about the command to set up the hook in the first place. Does windows have soft or hard links?

Reply:
Windows does both, `mklink Link Target` for a soft link at a file, `mklink /D Link Target` for a soft link at a directory, `mklink /H Link Target` for a hard link at a file and `mklink /J Link Target` for a hard link at a directory

Reply:
> Also, does it really need to be in a subdirectory if it's only one file?

If anything I would rename this to `.git_hooks` to remove it from view since it's not something most people will need to touch.